### ISSUE TITLE
Error trying to merge 3 .fit files - 

### ISSUE THREAD

**Issue by chris-obermeier** [sumoDave/GarminDiveFitAppender#4]
I'm trying to merge 3 files but the GarminDiveFITTool throws the following errors.

Attached are the zipped 3 fit files I'm trying to merge: [Fit Files.zip](https://github.com/user-attachments/files/15860994/Fit.Files.zip)

```
Processing file: \Fit Files\15949026647_ACTIVITY.fit
Processing file: \Fit Files\15949026648_ACTIVITY.fit
Processing file: \Fit Files\15949026650_ACTIVITY.fit
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because the return value of "com.garmin.fit.DiveSummaryMesg.getDiveNumber()" is null
Processing file: C:\Users\ChrisObermeier\Desktop\Fit Files\15949026647_ACTIVITY.fit
Processing file: C:\Users\ChrisObermeier\Desktop\Fit Files\15949026648_ACTIVITY.fit
Processing file: C:\Users\ChrisObermeier\Desktop\Fit Files\15949026650_ACTIVITY.fit
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because the return value of "com.garmin.fit.DiveSummaryMesg.getDiveNumber()" is null
        at DiveProfile.setDiveNumberCnsN2O2Values(DiveProfile.java:88)
        at DiveProfile.initialiseData(DiveProfile.java:22)
        at Main.main(Main.java:29)
```

**Comment by sumoDave** [comment]
Thanks for attaching the files . Seems that Order of Dive Summary messages have changed in fit files generated by the MK3. 
Will have a fix in soon.

**Comment by chris-obermeier** [comment]
Thank you!


Thanks,

Chris Obermeier

Head of Engineering

[Logo  Description automatically generated]

—

Remote

Email: ***@***.******@***.***>

Website: xpanse.com

Phone: 414.429.4185



[A picture containing company name  Description automatically generated]    [A picture containing icon  Description automatically generated]

________________________________
From: sumoDave ***@***.***>
Sent: Sunday, June 16, 2024 10:54:04 PM
To: sumoDave/GarminDiveFitAppender ***@***.***>
Cc: Chris Obermeier ***@***.***>; Author ***@***.***>
Subject: **EXT**Re: [sumoDave/GarminDiveFitAppender] Error trying to merge 3 .fit files - (Issue #4)


Thanks for attaching the files . Seems that Order of Dive Summary messages have changed in fit files generated by the MK3.
Will have a fix in soon.

**Comment by sumoDave** [comment]
Fixed in release v2.0.0
